internal/allocator/allocator.go

```diff
@@ -389,42 +434,49 @@ func sharingOK(existing, new *key) error {
 	return nil
 }
 
-// poolCount returns the number of addresses in the pool.
-func poolCount(p *config.Pool) int64 {
-	var total int64
+// poolCount returns the total number of IPv4 and IPv6 addresses in the pool.
+func poolCount(p *config.Pool) (int64, int64) {
+	var totalIPv4 int64
+	var totalIPv6 int64
+
 	for _, cidr := range p.CIDR {
 		o, b := cidr.Mask.Size()
-		if b-o >= 62 {
+		if b-o >= 62 || (cidr.IP.To4() == nil && totalIPv6 == math.MaxInt64) {
```

Comment:
Can we have this separated? Either as a separate if + continue or below, when we set totalIPv6?
It's gonna be clearer that we set it when we find it and from that point we start ignoring v6